parsons/etl_helpers/conversions.py

```diff
@@ -0,0 +1,188 @@
+on their own.
+"""
+
+from dateutil.parser import parse
+import datetime
+import re
+
+
+# Datetime conversion functions
+
+
+def date_to_timestamp(value, tzinfo=datetime.timezone.utc):
+    """Convert any date value into a Unix timestamp.
+
+    `Args:`
+        value: int or str or datetime
+            Value to parse
+        tzinfo: datetime.timezone
+            `Optional`: Timezone for the datetime; defaults to UTC.
+    `Returns:`
+        Unix timestamp (int)
+    """
+
+    parsed_date = parse_date(value)
+
+    if not parsed_date:
+        return None
+
+    if not parsed_date.tzinfo:
+        parsed_date = parsed_date.replace(tzinfo=tzinfo)
+
+    return int(parsed_date.timestamp())
+
+
+def parse_date(value, tzinfo=datetime.timezone.utc):
+    """Parse an arbitrary date value into a Python datetime.
+
+    If no value is provided (i.e., the value is None or empty), then the return value will be
+    None.
+
+    `Args:`
+        value: int or str or datetime
+            Value to parse
+        tzinfo: datetime.timezone
+            `Optional`: Timezone for the datetime; defaults to UTC.
+    `Returns:`
+        datetime.datetime or None
+    """
+
+    if not value:
+        return None
+
+    # If it's a number, we (probably) have a unix timestamp
+    if isinstance(value, int):
+        parsed = datetime.datetime.fromtimestamp(value, tzinfo)
+    elif isinstance(value, datetime.datetime):
+        parsed = value
+    elif isinstance(value, str):
+        parsed = parse(value)
+    else:
+        raise TypeError(
+            'Unable to parse value; must be one of string or int or datetime, but got type '
+            f'{type(value)}')
+
+    if not parsed.tzinfo:
+        parsed = parsed.replace(tzinfo=tzinfo)
+
+    return parsed
+
+
+def timestamp_to_readable(value, tzinfo=datetime.timezone.utc, format_as='%Y-%m-%d %H:%M:%S'):
+    """Converts a Unix timestamp into a common human-readable format.
+
+    `Args:`
+        value: int or str or datetime
+            Must correspond to a unix timestamp
+        tzinfo: datetime.timezone
+            `Optional`: Timezone for the datetime; defaults to UTC.
+        format_as: str
+            Should use strftime syntax; defaults to '%Y-%m-%d %H:%M:%S'
+    `Returns:`
+        str
+    """
+    datetime_obj = parse_date(value, tzinfo)
+    return datetime_obj.strftime(format_as)
+
+
+# Flatten contacts
+
+def get_primary_contact_from_nested(contact_list, get_first=True, selector=None):
+    """Extracts single contact value from list of dictionaries.
+
+    Multiple connectors, including Mobilize and Action Network, have a format for
+    contact info that looks like:
+
+    [{"primary": True, "number": "111-111-1111"}]
+
+    This helper method helps "flatten" the dictionary by returning the primary number (or,
+    if no primary number is found and get_first is True, the first number found).
```

Comment:
Idea: what if we made the keys of the dictionary a tuple with all the numbers? e.g. 
```
result = {
   ('5554444444'): {'foo': 'bar'},
   ('4441234567', '7771234567'): {'bin': 'ban'},
   # ...
}
```

I bring this up, because there would be no exceptional cases here. If users wanted to get the first number, they would simply access the tuple and get the first element.